hs_core/hydroshare/hs_bagit.py

```diff
@@ -222,6 +222,8 @@ def create_bag(resource):
 
     istorage.setAVU(resource.short_id, "resourceType", resource._meta.object_name)
 
+    # delete if there exists any bags for the resource
+    resource.bags.all().delete()
```

Comment:
@hyi The bulk ```delete()``` function of the QuerySet object I believe internally does what you have suggested. I will double check though.